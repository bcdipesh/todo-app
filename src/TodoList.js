import { useState } from "react";
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";

import "./TodoList.css";

function TodoList() {
  const INITIAL_STATE = { "todo-task": "" };
  const [formState, setFormState] = useState(INITIAL_STATE);
  const [todoList, setTodoList] = useState([]);

  function handleChange(e) {
    setFormState({
      ...formState,
      [e.target.name]: e.target.value,
    });
  }

  function handleSubmit(e) {
    e.preventDefault();
    setTodoList([...todoList, { task: formState["todo-task"] }]);
    setFormState(INITIAL_STATE);
  }

  function handleRemove(key) {
    setTodoList(todoList.filter((todoItem, idx) => idx !== key));
  }

  return (
    <div className="TodoList">
      <NewTodoForm handleChange={handleChange} handleSubmit={handleSubmit} />
      {todoList.length !== 0
        ? todoList.map(function (todoItem, idx) {
            return (
              <Todo
                key={idx}
                task={todoItem.task}
                handleRemove={() => handleRemove(idx)}
              />
            );
          })
        : ""}
    </div>
  );
}

export default TodoList;
