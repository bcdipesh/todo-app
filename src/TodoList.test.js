import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

function addTodo(todoList, task = "testing") {
  const todoTaskInput = todoList.getByLabelText("Todo Task:");
  fireEvent.change(todoTaskInput, { target: { value: task } });
  const addTodoBtn = todoList.getByText("Add Todo");
  fireEvent.click(addTodoBtn);
}

it("renders without crashing", function () {
  render(<TodoList />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("can add a todo", function () {
  const todoList = render(<TodoList />);
  addTodo(todoList);

  expect(todoList.getByText("testing")).toBeInTheDocument();
  expect(todoList.getByText("X")).toBeInTheDocument();
});

it("can remove a todo", function () {
  const todoList = render(<TodoList />);
  addTodo(todoList);

  fireEvent.click(todoList.getByText("X"));
  expect(todoList.queryByText("testing")).not.toBeInTheDocument();
});
