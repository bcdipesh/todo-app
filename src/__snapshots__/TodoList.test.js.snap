// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <div
      class="NewTodoForm"
    >
      <form>
        <label
          for="todo-task"
        >
          Todo Task: 
        </label>
        <input
          class="NewTodoForm-todo-task"
          id="todo-task"
          name="todo-task"
          placeholder="Add task..."
          type="text"
        />
        <button
          class="NewTodoForm-add-todo-btn"
          type="submit"
        >
          Add Todo
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
